packages/js/product-editor/src/blocks/variation-items/edit.tsx

```diff
@@ -19,11 +35,160 @@ export function Edit( {
 		isInSelectedTab?: boolean;
 	};
 } ) {
+	const noticeDimissed = useRef( false );
+	const { invalidateResolution } = useDispatch(
+		EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
+	);
+	const productId = useEntityId( 'postType', 'product' );
 	const blockProps = useBlockProps();
+	const [ productStatus ] = useEntityProp< string >(
+		'postType',
+		'product',
+		'status'
+	);
+
+	const totalCountWithoutPriceRequestParams = useMemo(
+		() => ( {
+			product_id: productId,
+			order: 'asc',
+			orderby: 'menu_order',
+			has_price: false,
+		} ),
+		[ productId ]
+	);
+
+	const { totalCountWithoutPrice } = useSelect(
+		( select ) => {
+			const { getProductVariationsTotalCount } = select(
+				EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
+			);
+
+			return {
+				totalCountWithoutPrice:
+					getProductVariationsTotalCount< number >(
+						totalCountWithoutPriceRequestParams
+					),
+			};
+		},
+		[ productId ]
+	);
+
+	const {
+		updateUserPreferences,
+		variable_items_without_price_notice_dismissed:
+			itemsWithoutPriceNoticeDismissed,
+	} = useUserPreferences();
+
+	const { ref: variationTableRef } = useValidation< Product >(
+		`variations`,
+		async function regularPriceValidator( defaultValue, additionalData ) {
+			if (
+				totalCountWithoutPrice > 0 &&
+				! noticeDimissed.current &&
+				productStatus !== 'publish' &&
+				additionalData?.status === 'publish'
```

Comment:
We cause a validation error if there is more then one variation without a price and if the product hasn't already published and if the notice hasn't been dismissed.

Reply:
Does having `additionalData?.status === 'publish'` means that it's "trying" to publish, or is it already published? It's not clear to me. Maybe we can rename the "status" to "nextStatus" or something like that?

Reply:
I will add a comment, and `additionalData?.status === 'publish'` means that it is publishing now.